rasa/shared/core/flows/flow.py

```diff
@@ -207,14 +287,54 @@ def from_json(flow_id: Text, flow_config: Dict[Text, Any]) -> Flow:
         Returns:
             The parsed flow.
         """
-        steps = flow_config.get("steps") or []
+        step_sequence = StepSequence.from_json(flow_config.get("steps"))
+
         return Flow(
             id=flow_id,
             name=flow_config.get("name", ""),
             description=flow_config.get("description"),
-            steps=[step_from_json(step_config) for step_config in steps],
+            step_sequence=Flow.resolve_default_ids(step_sequence),
         )
 
+    @staticmethod
+    def resolve_default_ids(step_sequence: StepSequence) -> StepSequence:
```

Comment:
Could we have some tests for this? 🙏🏻

Reply:
yes for sure, tests for this pr are still tbd - wanted to get feedback on the implementation first